src/_pytest/monkeypatch.py

```diff
@@ -225,7 +225,7 @@ def delenv(self, name, raising=True):
         If ``raising`` is set to False, no exception will be raised if the
         environment variable is missing.
         """
-        self.delitem(os.environ, name, raising=raising)
+        self.delitem(os.environ, str(name), raising=raising)
```

Comment:
it shouldn't matter for `delenv` right -- this doesn't store anything? (oh unless this causes it to put back the key with a unicode string?)

Reply:
Please see https://github.com/tox-dev/tox/pull/1025#discussion_r221273830, the bug was found actually with `delenv`. 😁

Reply:
@asottile the name is used in the lower level apis to remove the key, so it matters